presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/BuiltInQueryPreparer.java

```diff
@@ -179,5 +180,19 @@ public boolean isExplainTypeValidate()
                     .map(option -> (ExplainType) option)
                     .anyMatch(explainType -> explainType.getType() == VALIDATE);
         }
+
+        @Override
+        public boolean isExplainTypeIO()
+        {
+            if (!(statement instanceof Explain)) {
+                return false;
+            }
+
+            return ((Explain) statement).getOptions()
+                    .stream()
+                    .filter(option -> option instanceof ExplainType)
+                    .map(option -> (ExplainType) option)
+                    .anyMatch(explainType -> explainType.getType() == IO);
```

Comment:
can we have more than one explain type option in the query?

Reply:
No. will remove anyMatch.